src/plugins/intel_gpu/src/graph/impls/ocl/strided_slice.cpp

```diff
@@ -141,51 +142,53 @@ struct strided_slice_impl : typed_primitive_impl_ocl<strided_slice> {
-                if (params.striding_params[1][dim] < 0)
-                    params.striding_params[1][dim] = std::max(out_shape[dim] + params.striding_params[1][dim], (int32_t)0);
-
-                params.striding_params[0][dim] = std::min(params.striding_params[0][dim], out_shape[dim]);
-                params.striding_params[1][dim] = std::min(params.striding_params[1][dim], out_shape[dim]);
-
-                auto& begin = params.striding_params[0][dim];
-                auto& end = params.striding_params[1][dim];
-                auto& stride = params.striding_params[2][dim];
-                bool is_clamp_begin = begin_org != begin;
-                bool is_clamp_end = end_org != end;
-                bool is_reverse = stride < 0;
-                // If begin > end && is_reverse, then we don't need to adjust begin/end values, the kernel will process it correctly
-                // However, in case of out-of-bounds begin/end values, it will be clamped, so we subtract 1 from each of them manually
-                // E.g. out_shape[dim] = 100; begin=10000; end=-10000; stride=-1
-                // clamp: begin=100; end=0;
-                // sub: begin=99; end=-1;
-                // If begin <= end, then we swap begin/end values and subtruct 1 from each of them
-                // E.g. out_shape[dim] = 100; begin=0; end=100; stride=-1
-                // swap: begin=100; end=0;
-                // sub: begin=99; end=-1;
-                // So the kernel will put the slices [99, 0] in reversed order as expected.
-                if (is_reverse) {
-                    if (begin <= end) {
+                auto begin = params.striding_params[0][dim];
+                auto end = params.striding_params[1][dim];
+                auto stride = params.striding_params[2][dim];
+
+                // Check out of bounds values for Clamping
+                auto check_out_of_bounds = [&](int32_t value) -> bool {
+                    auto size = out_shape[dim];
+                    if (value >= size || value < (size * -1))
+                        return true;
+                    else
+                        return false;
+                };
+                bool should_clamp_begin = check_out_of_bounds(begin);
+                bool should_clamp_end = check_out_of_bounds(end);
+
+                // Convert a negative value which means reverse indexing from the end
+                if (begin < 0)
+                    begin += out_shape[dim];  // converted value can be negative if the original one was out of bounds
+                if (end < 0)
+                    end += out_shape[dim];
+                bool is_stride_reverse = (stride < 0) ? true : false;
+
+                // Clamping
+                if (should_clamp_begin)
+                    begin = std::min(std::max(begin, (int32_t)0), out_shape[dim]-1);
+                if (should_clamp_end)
+                    end = std::min(std::max(end, (int32_t)0), out_shape[dim]-1);
```

Comment:
if `out_shape[dim] = 100`, `begin=0`, `end=100`... this clamping will change `end` to `99`. Then only 99 elements will be copied, while original parameter was to copy 100 elements. Then I think this is not an intended behavior. Is my understanding correct?

Reply:
Yes, i will update the logic